### DISCUSSION TITLE
strings and sa.text() have a different behavior in server_default

### DISCUSSION THREAD

**Discussion by bfontaine** [sqlalchemy/alembic#1433] [Usage Questions] [answered]
Hello,

I use Alembic 1.3.1 (latest version) and the `postgres:14.7` image.

```python
from alembic import op
import sqlalchemy as sa

revision = 'ab743bfca885'
down_revision = None
branch_labels = None
depends_on = None

def upgrade():
    op.create_table(
        'test_alembic',
        sa.Column('with_sa_text', sa.DateTime(), server_default=sa.text('now()'), nullable=False),
        sa.Column('with_str',     sa.DateTime(), server_default='now()',          nullable=False),
    )

def downgrade():
    op.drop_table('test_alembic')
```

If I execute the above migration with `alembic alembic upgrade head` and then run `\d test_alembic` in `psql`, this is what I get:

```
                                                  Table "public.test_alembic"
    Column    |            Type             | Collation | Nullable |                          Default                          
--------------+-----------------------------+-----------+----------+-----------------------------------------------------------
 with_sa_text | timestamp without time zone |           | not null | now()
 with_str     | timestamp without time zone |           | not null | '2024-02-27 17:18:54.729436'::timestamp without time zone
```

The two columns have a different default: the one using `sa.text` has `NOW()`, while the one using a plain string has *the result* of running `NOW()` when we create the table.

I haven’t seen anything about this in the docs; is it a bug?

Thanks in advance.

**Comment by zzzeek** [comment, marked as answer]
no bug, if you send a string for a server_default, that's a literal default, just like if you sent an integer or a float.  docs are here https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.params.server_default
